tests/ut/ir/test_distributed_compiled_program.py

```
@@ -108,5 +114,93 @@ def test_call_validates_device_tensor_shape(compiled):
+    run_py = tmp_path / "debug" / "run.py"
+    if not run_py.exists():
+        pytest.skip("debug/run.py not emitted for this program")
+    sentinel = "# hand-edited by the user — must survive from_dir\n"
+    run_py.write_text(sentinel)
+    DistributedCompiledProgram.from_dir(tmp_path)
+    assert run_py.read_text() == sentinel
+
+
+def test_from_dir_missing_meta_raises(tmp_path):
+    """A directory without distributed_meta.json raises with a recompile hint."""
+    with pytest.raises(FileNotFoundError, match="distributed_meta.json"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)